src/main/java/com/iota/iri/service/validation/impl/TransactionSolidifierImpl.java

```diff
@@ -0,0 +1,383 @@
+package com.iota.iri.service.validation.impl;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.iota.iri.controllers.TipsViewModel;
+import com.iota.iri.controllers.TransactionViewModel;
+import com.iota.iri.model.Hash;
+import com.iota.iri.network.pipeline.TransactionProcessingPipeline;
+import com.iota.iri.network.TransactionRequester;
+import com.iota.iri.service.snapshot.SnapshotProvider;
+import com.iota.iri.service.validation.TransactionSolidifier;
+import com.iota.iri.storage.Tangle;
+import com.iota.iri.utils.log.interval.IntervalLogger;
+import com.iota.iri.utils.thread.DedicatedScheduledExecutorService;
+import com.iota.iri.utils.thread.SilentScheduledExecutorService;
+
+import java.util.*;
+import java.util.concurrent.*;
+
+import static com.iota.iri.controllers.TransactionViewModel.PREFILLED_SLOT;
+import static com.iota.iri.controllers.TransactionViewModel.fromHash;
+
+/**
+ * A solidifier class for processing transactions. Transactions are checked for solidity, and missing transactions are
+ * subsequently requested. Once a transaction is solidified correctly it is placed into a broadcasting set to be sent to
+ * neighboring nodes.
+ */
+public class TransactionSolidifierImpl implements TransactionSolidifier {
+
+    private Tangle tangle;
+    private SnapshotProvider snapshotProvider;
+    private TransactionRequester transactionRequester;
+
+    /**
+     * Max size for all queues.
+     */
+    private static final int MAX_SIZE= 10000;
+
+    private static final int SOLIDIFICATION_INTERVAL = 100;
+
+    private static final IntervalLogger log = new IntervalLogger(TransactionSolidifier.class);
+
+    /**
+     * Executor service for running the {@link #processTransactionsToSolidify()}.
+     */
+    private SilentScheduledExecutorService executorService = new DedicatedScheduledExecutorService(
+            "Transaction Solidifier", log.delegate());
+
+    /**
+     * A queue for processing transactions with the {@link #checkSolidity(Hash)} call. Once a transaction has been
+     * marked solid it will be placed into the {@link #transactionsToBroadcast} queue.
+     */
+    private BlockingQueue<Hash> transactionsToSolidify = new ArrayBlockingQueue<>(MAX_SIZE);
```

Comment:
I remember I told you to use ArrayBlockingQueue...
We have a bug in networking #1742, that is fixed by doing this https://github.com/GalRogozinski/iri/commit/f65e29b23f89a8422aa5949d5d189080507c68b6

What happens is that sometiems the queue in networking gets filled in high tps regimes...
So I am rethinking my recommendation

Because before a memory attack IRI just might stop functioning...

Reply:
I always wondered about that but wasn't sure if i was thinking straight about it. Should I switch this to a `LinkedBlockingQueue` instead?